### DISCUSSION TITLE
How to apply identical displacements to both ends of a rectangular mesh

### DISCUSSION THREAD

**Discussion by zzzmx-josh** [idaholab/moose#27599] [Q&A General] [answered]
### Check these boxes if you have followed the posting rules.

- [X] Q&A General is the most appropriate section for my question
- [X] I have consulted the posting Guidelines on the Discussions front page
- [X] I have searched the Discussions forum and my question has not been asked before
- [X] I have searched the MOOSE website and the documentation does not answer my question
- [X] I have formatted my post following the posting guidelines (screenshots as a last resort, triple back quotes around pasted text)

### Question

How can I apply identical displacements at both ends of a rectangular grid to create an arch-shaped deformation? 
Do I need to establish a reference point with a displacement of 0? If so, where should this reference point be set?
Here is the schematic diagram: 
![image](https://github.com/idaholab/moose/assets/95216887/7da94b0e-4b6f-4485-9a6d-66357d6888d6)

**Comment by GiudGiud** [comment]
Hello

Use a FunctionDirichletBC on both ends and slowly apply the displacement if you want to see the strain appear.
The transformation of the surface can be derived analytically if it's just an angle. 

Guillaume

**Reply by zzzmx-josh** [reply]
Do I need to fix the displacement of a reference point to zero when using a FunctionDirichletBC on both ends?
Currently, I've attempted to directly apply the ADNeumannBC boundary condition on the left and right boundaries and specified disp_z. I've used a steady solver, but it's not converging at all.

**Reply by GiudGiud** [reply]
NeumannBC is not used to set a displacement, it's used to impose a force. You should use the Pressure action for that purpose.
If you only impose forces on the two ends, you do need to add dirichletBCs to remove extraneous translation / rotation dofs

**Reply by zzzmx-josh** [reply]
I made a mistake previously; I applied force using the NeumannBC. My mesh is 3D. How should I eliminate or constrain the excess degrees of freedom to achieve convergence? Currently, my boundary conditions are set as follows:

[BCs]
  [Load1]
    type = ADNeumannBC
    boundary = right
    value = 0.05
    variable = disp_z
  []
  [Load2]
    type = ADNeumannBC
    boundary = left
    value = 0.05
    variable = disp_z
  []
[]

**Reply by GiudGiud** [reply, marked as answer]
You could pick a surface in the mesh and apply 0 displacements there. For example the symmetry plane of the mesh?
If you are not going to constrain the disaplacement of the surface edges, this would work

**Reply by zzzmx-josh** [reply]
Exactly, I don't need to restrict displacement.

**Reply by zzzmx-josh** [reply]
Thanks for your advice
